string ir_left
string ir_right
int32 ir_left_stat
int32 ir_right_stat
